src/repositories/live.repository.ts

```diff
@@ -0,0 +1,330 @@
+import { LiveLikeCountResponseRootObject } from '../responses/live.like-count.response';
+import { LiveJoinRequestCountsResponseRootObject } from '../responses/live.join-request-counts.response';
+
+export class LiveRepository extends Repository {
+
+  public async disableComments(broadcastId: string): Promise<LiveSwitchCommentsResponseRootObject> {
+    const { body } = await this.client.request.send<LiveSwitchCommentsResponseRootObject>({
+      url: `/api/v1/live/${broadcastId}/mute_comment/`,
+      method: 'POST',
+      form: this.client.request.sign({
+        _csrftoken: this.client.state.cookieCsrfToken,
+        _uid: this.client.state.cookieUserId,
+        _uuid: this.client.state.uuid,
+      }),
+    });
+    return body;
+  }
+
+  public async getComments(broadcastId: string, commentsRequested: number = 4, lastCommentTs: string = ''): Promise<LiveCommentsResponseRootObject> {
+    const { body } = await this.client.request.send<LiveCommentsResponseRootObject>({
+      url:
+        lastCommentTs === '' ?
+          `/api/v1/live/${broadcastId}/get_comment/?num_comments_requested=${commentsRequested}`
+          : `/api/v1/live/${broadcastId}/get_comment/?last_comment_ts=${lastCommentTs}&num_comments_requested=${commentsRequested}`,
+      method: 'GET',
+    });
+    return body;
+  }
+
+  public async getHeartbeatViewerCount(broadcastId: string): Promise<LiveHeartbeatViewerCountResponseRootObject> {
+    const { body } = await this.client.request.send<LiveHeartbeatViewerCountResponseRootObject>({
+      url: `/api/v1/live/${broadcastId}/heartbeat_and_get_viewer_count/`,
+      form: {
+        _csrftoken: this.client.state.cookieCsrfToken,
+        offset_to_video_start: 0,
+        _uuid: this.client.state.uuid,
+      },
+      method: 'POST',
+    });
+    return body;
+  }
+
+  public async getInfo(broadcastId: string): Promise<LiveInfoResponseRootObject> {
+    const { body } = await this.client.request.send<LiveInfoResponseRootObject>({
+      url: `/api/v1/live/${broadcastId}/info/`,
+      method: 'GET',
+    });
+    return body;
+  }
+
+  public async getFinalViewers(broadcastId: string): Promise<LiveFinalViewersResponseRootObject> {
+    const { body } = await this.client.request.send<LiveFinalViewersResponseRootObject>({
+      url: `api/v1/live/${broadcastId}/get_final_viewer_list/`,
+      method: 'GET',
+    });
+    return body;
+  }
+
+  public async enableComments(broadcastId: string): Promise<LiveSwitchCommentsResponseRootObject> {
+    const { body } = await this.client.request.send<LiveSwitchCommentsResponseRootObject>({
+      url: `/api/v1/live/${broadcastId}/unmute_comment/`,
+      method: 'POST',
+      form: this.client.request.sign({
+        _csrftoken: this.client.state.cookieCsrfToken,
+        _uid: this.client.state.cookieUserId,
+        _uuid: this.client.state.uuid,
+      }),
+    });
+    return body;
+  }
+
+  public async createBroadcast(
```

Comment:
When you have more than one optional input parameter you should accept it as a single object, because if user need to specify one `message` the call in this case wil look like `createBroadcast(void 0, void 0, 'Hello world')`
Please, refactor it in al the places.